controllers/messages/MessageController.ts

```diff
@@ -120,9 +121,13 @@ export default class MessageController implements IMessageController {
     // Emit to client sockets
     const recipients = newMessage.conversation.participants;
     for (const recipient of recipients) {
-      const newNotification: Notification = await this.notificationDao.createNotificationForUser("MESSAGES", recipient, req.params.userId,);
       this.socketServer.to(recipient.toString()).emit('NEW_MESSAGE', message);
-      this.socketServer.to(recipient.toString()).emit('NEW_NOTIFICATION', message);
+      if (recipient.toString() === req.params.userId) {
+        continue;
+      } else {
+        const newNotification: Notification = await this.notificationDao.createNotificationForUser("MESSAGES", recipient, req.params.userId);
+        this.socketServer.to(recipient.toString()).emit('NEW_NOTIFICATION', newNotification);
```

Comment:
I'm not sure why toString() is being called on recipient here. Has this been tested? The goal is to emit to a user id, so you need be careful not be emitting to string of an entire user object